apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -78,4 +115,81 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  @doc """
+  A function that adds a merkle proof for every single transaction
+  """
+
+  @spec add_proof_to_txs(list()) :: list()
+  def add_proof_to_txs(user_txs) do
```

Comment:
what is `user_txs` a list of? are these `%SignedTx{}` objects?
* If so, can you specify that in the spec? we will also need to update the existing [SignedTx](https://github.com/aeternity/elixir-research/blob/master/apps/aecore/lib/aecore/structures/signed_tx.ex) structure with the appropriate fields (tx.txs_hash, tx.data, etc..)
* If not, what structure is this? Does it need to be defined for this PR?

Reply:
It returns the merkle proof for each transaction as api endpoint to the wallet, so the wallet can in theory verify them

Reply:
Internally we already discussed that the code structure could potentially be optimized but come to an agreement that its good enough for now